File: q2/q2.py
numMinions = int(input())
list = []

index = 0;
while (index < numMinions):
    line = input()
    nums = line.split(' ')
    low = int(nums[0])
    high = int(nums[1])
    list.append((low, high))
    index = index+1

sortL = sorted(list, key=lambda x: x[0])
sortL = sorted(sortL, key=lambda x: x[1])

counter = 1
thres = sortL[0][1]
            
for i in range(len(sortL)):
    if sortL[i][0] <= thres:
        if sortL[i][1] <= thres:
            thres = sortL[i][1]
        continue
    counter = counter+1
    thres = sortL[i][1]
                        
print(counter)

File: q2/q2.js
var line = readline();
var numMinions = parseInt(line);
var list = [];

var index = 0;
while (index < numMinions) {
    line = readline();
    var nums = line.split(' ');
    var low = parseInt(nums[0]);
    var high = parseInt(nums[1]);
    list.push({x:low, y:high});
    index++;
}

list.sort(function(a,b){
    if (a.x < b.x) return -1;
    if (a.x > b.x) return 1;
    if (a.y < b.y) return -1;
    if (a.y > b.y) return 1;
    return 0;
});

var counter = 1;
var thres = list[0].y;

for (var i = 0;i<list.length;i++){
    if (list[i].x <= thres) {
        if (list[i].y <= thres) {
            thres = list[i].y;
        }
        continue;
    }
    counter++;
    thres = list[i].y;
    
}
print(counter);
